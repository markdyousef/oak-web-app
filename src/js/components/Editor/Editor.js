import React, { Component, PropTypes } from 'react';

class Editor extends Component {
    constructor() {
        super();
        this.state = {};
    }
    render() {
        return (
            <div>
                EDITOR
            </div>
        );
    }
}

export default Editor;
