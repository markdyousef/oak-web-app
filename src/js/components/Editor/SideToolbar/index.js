import createSideToolbarPlugin from 'draft-js-side-toolbar-plugin';
import buttonStyles from './buttonStyles.css';
import toolbarStyles from './toolbarStyles.css';
import blockTypeSelectStyles from './blockTypeSelectStyles.css';

const sideToolbarPlugin = createSideToolbarPlugin({
    theme: { buttonStyles, toolbarStyles, blockTypeSelectStyles }
});


export default sideToolbarPlugin;
