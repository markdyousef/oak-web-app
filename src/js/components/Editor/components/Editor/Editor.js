// @flow
import React, { Component } from 'react';
import { RichUtils, Editor } from 'draft-js';
// TODO: add custom styling
import 'draft-js/dist/Draft.css';
import styled from 'styled-components';
import Toolbar from '../Toolbar';
import FloatingActionButton from '../FloatingActionButton';
import customRenderer from '../../utils/customRenderer';
import { Block } from '../../utils/constants';

const Container = styled.div`
    ${''/* padding: 30px 30px; */}
    box-sizing: border-box;
    position: relative;
    & h1 {
        font-size: 48px;
        font-weight: bold;
        letter-spacing: -.024em;
        line-height: 1.18;
        margin-bottom: 20px;
        color: #131517;

    }
    & h2 {
        font-size: 28px;
        font-weight: normal;
        letter-spacing: -.008em;
        line-height: 1.24;
        margin-bottom: 20px;
        color: #797C80;
    }
    & ul {
        padding-left: 0;
        list-style: none;
    }
    & ol {
        padding-left: 0;
        list-style: none;
    }
    & li {
        font-size: 21px;
        line-height: 1,78;
    }
`;

const EditorContainer = styled.div`
    cursor: text;
    position: relative;
    margin: 0 auto;
    margin-top: 10px;
    max-width: 700px;
`;

type DefaultProps = {
    editorState: Object,
    canEdit: false,
    onChange: Function
}

type Props = {
    editorState: Object,
    canEdit: boolean,
    onChange: Function
}

type State = {
    editorState: Object
}

export default class App extends Component {
    defaultProps: DefaultProps
    props: Props
    state: State
    onTab = (event:Object) => {
        const { onChange, editorState } = this.props;
        // depth on ul and ol
        const levels = 2;
        const newEditorState = RichUtils.onTab(event, editorState, levels);
        if (newEditorState !== editorState) {
            onChange(newEditorState);
        }
    };
    getEditorState = () => this.state.editorState;
    focus = () => this.editor.focus();
    toggleBlockType = (blockType:Object) => {
        const { onChange, editorState } = this.props;
        const type = RichUtils.getCurrentBlockType(editorState);
        if (type.indexOf(`${Block.ATOMIC}:`) === 0) return;
        onChange(
            RichUtils.toggleBlockType(
                editorState,
                blockType
            )
        );
    }
    toggleInlineStyle = (inlineStyle:Object) => {
        const { onChange, editorState } = this.props;
        onChange(
            RichUtils.toggleInlineStyle(
                editorState,
                inlineStyle
            )
        );
    }
    render() {
        const { canEdit, editorState, onChange } = this.props;
        const showToolbar = !editorState.getSelection().isCollapsed();
        return (
            <Container>
                <EditorContainer>
                    <Editor
                        ref={(node) => { this.editor = node; }}
                        editorState={editorState}
                        spellCheck
                        placeholder="Write something cool..."
                        onChange={onChange}
                        blockRendererFn={customRenderer(editorState, onChange)}
                        onTab={this.onTab}
                        readOnly={!canEdit}
                    />
                    {canEdit && <FloatingActionButton
                        editorState={editorState}
                        focus={this.focus}
                        setEditorState={onChange}
                    />}
                    <Toolbar
                        editorState={editorState}
                        toggleBlockType={this.toggleBlockType}
                        toggleInlineStyle={this.toggleInlineStyle}
                        focus={this.focus}
                        editorNode={this.editor}
                        showToolbar={showToolbar}
                    />
                </EditorContainer>
            </Container>
        );
    }
}
