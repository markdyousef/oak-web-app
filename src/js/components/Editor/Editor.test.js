import React from 'react';
import { shallow } from 'enzyme';
import Editor from './Editor';

it('renders correctly', () => {
    const props = {
    };
    const shallowComponent = shallow(
        <Editor {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
