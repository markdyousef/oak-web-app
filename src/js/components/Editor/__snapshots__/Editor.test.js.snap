exports[`test renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    className={undefined}
    onClick={[Function]}>
    <Decorated(Toolbar) />
    <PluginEditor
        customStyleMap={Object {}}
        decorators={Array []}
        defaultBlockRenderMap={true}
        defaultKeyBindings={true}
        editorState={
            EditorState {
                "_immutable": Object {
                  "allowUndo": true,
                  "currentContent": Object {
                    "blockMap": Object {
                      "evpha": Object {
                        "characterList": Array [],
                        "data": Object {},
                        "depth": 0,
                        "key": "evpha",
                        "text": "",
                        "type": "unstyled",
                      },
                    },
                    "entityMap": Object {
                      "__add": [Function],
                      "__create": [Function],
                      "__get": [Function],
                      "__getLastCreatedEntityKey": [Function],
                      "__mergeData": [Function],
                      "__replaceData": [Function],
                      "add": [Function],
                      "create": [Function],
                      "get": [Function],
                      "getLastCreatedEntityKey": [Function],
                      "mergeData": [Function],
                      "replaceData": [Function],
                    },
                    "selectionAfter": Object {
                      "anchorKey": "evpha",
                      "anchorOffset": 0,
                      "focusKey": "evpha",
                      "focusOffset": 0,
                      "hasFocus": false,
                      "isBackward": false,
                    },
                    "selectionBefore": Object {
                      "anchorKey": "evpha",
                      "anchorOffset": 0,
                      "focusKey": "evpha",
                      "focusOffset": 0,
                      "hasFocus": false,
                      "isBackward": false,
                    },
                  },
                  "decorator": null,
                  "directionMap": Object {
                    "evpha": "LTR",
                  },
                  "forceSelection": false,
                  "inCompositionMode": false,
                  "inlineStyleOverride": null,
                  "lastChangeType": null,
                  "nativelyRenderedContent": null,
                  "redoStack": Array [],
                  "selection": Object {
                    "anchorKey": "evpha",
                    "anchorOffset": 0,
                    "focusKey": "evpha",
                    "focusOffset": 0,
                    "hasFocus": false,
                    "isBackward": false,
                  },
                  "treeMap": Object {
                    "evpha": Array [
                      Object {
                        "decoratorKey": null,
                        "end": 0,
                        "leaves": Array [
                          Object {
                            "end": 0,
                            "start": 0,
                          },
                        ],
                        "start": 0,
                      },
                    ],
                  },
                  "undoStack": Array [],
                },
              }
        }
        onChange={[Function]}
        plugins={
            Array [
                Object {
                  "SideToolbar": [Function],
                  "initialize": [Function],
                  "onChange": [Function],
                },
              ]
        } />
</styled.div>,
  "nodes": Array [
    <styled.div
      className={undefined}
      onClick={[Function]}>
      <Decorated(Toolbar) />
      <PluginEditor
            customStyleMap={Object {}}
            decorators={Array []}
            defaultBlockRenderMap={true}
            defaultKeyBindings={true}
            editorState={
                  EditorState {
                        "_immutable": Object {
                          "allowUndo": true,
                          "currentContent": Object {
                            "blockMap": Object {
                              "evpha": Object {
                                "characterList": Array [],
                                "data": Object {},
                                "depth": 0,
                                "key": "evpha",
                                "text": "",
                                "type": "unstyled",
                              },
                            },
                            "entityMap": Object {
                              "__add": [Function],
                              "__create": [Function],
                              "__get": [Function],
                              "__getLastCreatedEntityKey": [Function],
                              "__mergeData": [Function],
                              "__replaceData": [Function],
                              "add": [Function],
                              "create": [Function],
                              "get": [Function],
                              "getLastCreatedEntityKey": [Function],
                              "mergeData": [Function],
                              "replaceData": [Function],
                            },
                            "selectionAfter": Object {
                              "anchorKey": "evpha",
                              "anchorOffset": 0,
                              "focusKey": "evpha",
                              "focusOffset": 0,
                              "hasFocus": false,
                              "isBackward": false,
                            },
                            "selectionBefore": Object {
                              "anchorKey": "evpha",
                              "anchorOffset": 0,
                              "focusKey": "evpha",
                              "focusOffset": 0,
                              "hasFocus": false,
                              "isBackward": false,
                            },
                          },
                          "decorator": null,
                          "directionMap": Object {
                            "evpha": "LTR",
                          },
                          "forceSelection": false,
                          "inCompositionMode": false,
                          "inlineStyleOverride": null,
                          "lastChangeType": null,
                          "nativelyRenderedContent": null,
                          "redoStack": Array [],
                          "selection": Object {
                            "anchorKey": "evpha",
                            "anchorOffset": 0,
                            "focusKey": "evpha",
                            "focusOffset": 0,
                            "hasFocus": false,
                            "isBackward": false,
                          },
                          "treeMap": Object {
                            "evpha": Array [
                              Object {
                                "decoratorKey": null,
                                "end": 0,
                                "leaves": Array [
                                  Object {
                                    "end": 0,
                                    "start": 0,
                                  },
                                ],
                                "start": 0,
                              },
                            ],
                          },
                          "undoStack": Array [],
                        },
                      }
            }
            onChange={[Function]}
            plugins={
                  Array [
                        Object {
                          "SideToolbar": [Function],
                          "initialize": [Function],
                          "onChange": [Function],
                        },
                      ]
            } />
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <MainEditor />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MainEditor {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "focus": [Function],
        "onChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "editorState": EditorState {
            "_immutable": Object {
              "allowUndo": true,
              "currentContent": Object {
                "blockMap": Object {
                  "evpha": Object {
                    "characterList": Array [],
                    "data": Object {},
                    "depth": 0,
                    "key": "evpha",
                    "text": "",
                    "type": "unstyled",
                  },
                },
                "entityMap": Object {
                  "__add": [Function],
                  "__create": [Function],
                  "__get": [Function],
                  "__getLastCreatedEntityKey": [Function],
                  "__mergeData": [Function],
                  "__replaceData": [Function],
                  "add": [Function],
                  "create": [Function],
                  "get": [Function],
                  "getLastCreatedEntityKey": [Function],
                  "mergeData": [Function],
                  "replaceData": [Function],
                },
                "selectionAfter": Object {
                  "anchorKey": "evpha",
                  "anchorOffset": 0,
                  "focusKey": "evpha",
                  "focusOffset": 0,
                  "hasFocus": false,
                  "isBackward": false,
                },
                "selectionBefore": Object {
                  "anchorKey": "evpha",
                  "anchorOffset": 0,
                  "focusKey": "evpha",
                  "focusOffset": 0,
                  "hasFocus": false,
                  "isBackward": false,
                },
              },
              "decorator": null,
              "directionMap": Object {
                "evpha": "LTR",
              },
              "forceSelection": false,
              "inCompositionMode": false,
              "inlineStyleOverride": null,
              "lastChangeType": null,
              "nativelyRenderedContent": null,
              "redoStack": Array [],
              "selection": Object {
                "anchorKey": "evpha",
                "anchorOffset": 0,
                "focusKey": "evpha",
                "focusOffset": 0,
                "hasFocus": false,
                "isBackward": false,
              },
              "treeMap": Object {
                "evpha": Array [
                  Object {
                    "decoratorKey": null,
                    "end": 0,
                    "leaves": Array [
                      Object {
                        "end": 0,
                        "start": 0,
                      },
                    ],
                    "start": 0,
                  },
                ],
              },
              "undoStack": Array [],
            },
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          className={undefined}
          onClick={[Function]}>
          <Decorated(Toolbar) />
          <PluginEditor
                    customStyleMap={Object {}}
                    decorators={Array []}
                    defaultBlockRenderMap={true}
                    defaultKeyBindings={true}
                    editorState={
                              EditorState {
                                        "_immutable": Object {
                                          "allowUndo": true,
                                          "currentContent": Object {
                                            "blockMap": Object {
                                              "evpha": Object {
                                                "characterList": Array [],
                                                "data": Object {},
                                                "depth": 0,
                                                "key": "evpha",
                                                "text": "",
                                                "type": "unstyled",
                                              },
                                            },
                                            "entityMap": Object {
                                              "__add": [Function],
                                              "__create": [Function],
                                              "__get": [Function],
                                              "__getLastCreatedEntityKey": [Function],
                                              "__mergeData": [Function],
                                              "__replaceData": [Function],
                                              "add": [Function],
                                              "create": [Function],
                                              "get": [Function],
                                              "getLastCreatedEntityKey": [Function],
                                              "mergeData": [Function],
                                              "replaceData": [Function],
                                            },
                                            "selectionAfter": Object {
                                              "anchorKey": "evpha",
                                              "anchorOffset": 0,
                                              "focusKey": "evpha",
                                              "focusOffset": 0,
                                              "hasFocus": false,
                                              "isBackward": false,
                                            },
                                            "selectionBefore": Object {
                                              "anchorKey": "evpha",
                                              "anchorOffset": 0,
                                              "focusKey": "evpha",
                                              "focusOffset": 0,
                                              "hasFocus": false,
                                              "isBackward": false,
                                            },
                                          },
                                          "decorator": null,
                                          "directionMap": Object {
                                            "evpha": "LTR",
                                          },
                                          "forceSelection": false,
                                          "inCompositionMode": false,
                                          "inlineStyleOverride": null,
                                          "lastChangeType": null,
                                          "nativelyRenderedContent": null,
                                          "redoStack": Array [],
                                          "selection": Object {
                                            "anchorKey": "evpha",
                                            "anchorOffset": 0,
                                            "focusKey": "evpha",
                                            "focusOffset": 0,
                                            "hasFocus": false,
                                            "isBackward": false,
                                          },
                                          "treeMap": Object {
                                            "evpha": Array [
                                              Object {
                                                "decoratorKey": null,
                                                "end": 0,
                                                "leaves": Array [
                                                  Object {
                                                    "end": 0,
                                                    "start": 0,
                                                  },
                                                ],
                                                "start": 0,
                                              },
                                            ],
                                          },
                                          "undoStack": Array [],
                                        },
                                      }
                    }
                    onChange={[Function]}
                    plugins={
                              Array [
                                        Object {
                                          "SideToolbar": [Function],
                                          "initialize": [Function],
                                          "onChange": [Function],
                                        },
                                      ]
                    } />
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          className={undefined}
          onClick={[Function]}>
          <Decorated(Toolbar) />
          <PluginEditor
                    customStyleMap={Object {}}
                    decorators={Array []}
                    defaultBlockRenderMap={true}
                    defaultKeyBindings={true}
                    editorState={
                              EditorState {
                                        "_immutable": Object {
                                          "allowUndo": true,
                                          "currentContent": Object {
                                            "blockMap": Object {
                                              "evpha": Object {
                                                "characterList": Array [],
                                                "data": Object {},
                                                "depth": 0,
                                                "key": "evpha",
                                                "text": "",
                                                "type": "unstyled",
                                              },
                                            },
                                            "entityMap": Object {
                                              "__add": [Function],
                                              "__create": [Function],
                                              "__get": [Function],
                                              "__getLastCreatedEntityKey": [Function],
                                              "__mergeData": [Function],
                                              "__replaceData": [Function],
                                              "add": [Function],
                                              "create": [Function],
                                              "get": [Function],
                                              "getLastCreatedEntityKey": [Function],
                                              "mergeData": [Function],
                                              "replaceData": [Function],
                                            },
                                            "selectionAfter": Object {
                                              "anchorKey": "evpha",
                                              "anchorOffset": 0,
                                              "focusKey": "evpha",
                                              "focusOffset": 0,
                                              "hasFocus": false,
                                              "isBackward": false,
                                            },
                                            "selectionBefore": Object {
                                              "anchorKey": "evpha",
                                              "anchorOffset": 0,
                                              "focusKey": "evpha",
                                              "focusOffset": 0,
                                              "hasFocus": false,
                                              "isBackward": false,
                                            },
                                          },
                                          "decorator": null,
                                          "directionMap": Object {
                                            "evpha": "LTR",
                                          },
                                          "forceSelection": false,
                                          "inCompositionMode": false,
                                          "inlineStyleOverride": null,
                                          "lastChangeType": null,
                                          "nativelyRenderedContent": null,
                                          "redoStack": Array [],
                                          "selection": Object {
                                            "anchorKey": "evpha",
                                            "anchorOffset": 0,
                                            "focusKey": "evpha",
                                            "focusOffset": 0,
                                            "hasFocus": false,
                                            "isBackward": false,
                                          },
                                          "treeMap": Object {
                                            "evpha": Array [
                                              Object {
                                                "decoratorKey": null,
                                                "end": 0,
                                                "leaves": Array [
                                                  Object {
                                                    "end": 0,
                                                    "start": 0,
                                                  },
                                                ],
                                                "start": 0,
                                              },
                                            ],
                                          },
                                          "undoStack": Array [],
                                        },
                                      }
                    }
                    onChange={[Function]}
                    plugins={
                              Array [
                                        Object {
                                          "SideToolbar": [Function],
                                          "initialize": [Function],
                                          "onChange": [Function],
                                        },
                                      ]
                    } />
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MainEditor />,
}
`;
